impeller/renderer/BUILD.gn

```diff
@@ -4,18 +4,84 @@
 
 import("//flutter/impeller/tools/impeller.gni")
 
-impeller_component("renderer") {
+impeller_component("allocation") {
```

Comment:
This does more than just allocation and contains stuff like formats and descriptors. Perhaps `//impeller/core` or `//impeller/device`? (I like the former).

Reply:
I was originally going to call this "buffers" but it also has textures and formats and an allocator. Right now it's a target in `//impeller/renderer` still. I'm fine naming it whatever, moving to core